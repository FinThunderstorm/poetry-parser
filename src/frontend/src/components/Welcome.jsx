import React from 'react'
import { useSelector } from 'react-redux'
import { Link } from 'react-router-dom'
import { Button, Paper, Stack, Typography } from '@mui/material'

/** Welcome shows welcoming words to application
 */
const Welcome = () => {
    const status = useSelector((state) => state.packages.status)

    if (status === 'succeeded' || status === 'loading') {
        return <></>
    }
    return (
        <Paper elevation={6} sx={{ p: 2 }}>
            <Stack spacing={2} alignItems="center">
                <Typography variant="button" sx={{ fontSize: '3rem' }}>
                    Welcome to poetry-parser!
                </Typography>
                <Typography>
                    Begin explorating your poetry.lock -file by uploading file.
                </Typography>
                <Button component={Link} to="/upload" variant="contained">
                    Upload
                </Button>
            </Stack>
        </Paper>
    )
}

export default Welcome
