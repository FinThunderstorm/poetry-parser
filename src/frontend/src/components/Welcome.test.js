/* eslint-disable react/prop-types */
import React from 'react'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect'
import { render, screen } from '@testing-library/react'
import { Provider } from 'react-redux'
import { MemoryRouter, Route, Routes } from 'react-router-dom'
import { configureStore } from '@reduxjs/toolkit'

import Welcome from './Welcome'

import packagesReducer from '../reducers/packagesSlice'

describe('Welcome', () => {
    const init = {
        packages: {
            packages: {},
            status: 'idle',
            error: null,
        },
    }

    const WelcomeWrapper = ({ url, preloadedState }) => {
        const store = configureStore({
            reducer: {
                packages: packagesReducer,
            },
            preloadedState,
        })
        return (
            <MemoryRouter initialEntries={[url]}>
                <Provider store={store}>
                    <Routes>
                        <Route path="/" element={<Welcome />} />
                    </Routes>
                </Provider>
            </MemoryRouter>
        )
    }

    test('Welcome renderes correctly', () => {
        init.packages.status = 'idle'
        render(<WelcomeWrapper url="/" preloadedState={init} />)

        const title = screen.getByText('Welcome to poetry-parser!', {
            exact: false,
        })
        const info = screen.getByText(
            'Begin explorating your poetry.lock -file by uploading file.',
            {
                exact: false,
            }
        )
        expect(title).toBeDefined()
        expect(info).toBeDefined()

        const tree = renderer
            .create(<WelcomeWrapper url="/" preloadedState={init} />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })

    test('Welcome shows if failed', () => {
        init.packages.status = 'failed'
        render(<WelcomeWrapper url="/" preloadedState={init} />)

        const title = screen.getByText('Welcome to poetry-parser!', {
            exact: false,
        })
        const info = screen.getByText(
            'Begin explorating your poetry.lock -file by uploading file.',
            {
                exact: false,
            }
        )
        expect(title).toBeDefined()
        expect(info).toBeDefined()

        const tree = renderer
            .create(<WelcomeWrapper url="/" preloadedState={init} />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })

    test('Welcome is not shown if packages occurs', () => {
        init.packages.status = 'succeeded'
        render(<WelcomeWrapper url="/" preloadedState={init} />)

        const name = screen.queryByText('Welcome to poetry-parser!', {
            exact: false,
        })
        expect(name).toBeNull()

        const tree = renderer
            .create(<WelcomeWrapper url="/" preloadedState={init} />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })

    test('Welcome is not shown if loading packages', () => {
        init.packages.status = 'loading'
        render(<WelcomeWrapper url="/" preloadedState={init} />)

        const name = screen.queryByText('Welcome to poetry-parser!', {
            exact: false,
        })
        expect(name).toBeNull()

        const tree = renderer
            .create(<WelcomeWrapper url="/" preloadedState={init} />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
