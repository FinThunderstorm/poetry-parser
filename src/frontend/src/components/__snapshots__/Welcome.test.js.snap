// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Welcome is not shown if loading packages 1`] = `null`;

exports[`Welcome Welcome is not shown if packages occurs 1`] = `null`;

exports[`Welcome Welcome renderes correctly 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation6 css-1p3yn26-MuiPaper-root"
>
  <div
    className="css-1y48f0p-MuiStack-root"
  >
    <span
      className="MuiTypography-root MuiTypography-button css-1xjj6hs-MuiTypography-root"
    >
      Welcome to poetry-parser!
    </span>
    <p
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Begin explorating your poetry.lock -file by uploading file.
    </p>
    <a
      className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
      href="/upload"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
    >
      Upload
    </a>
  </div>
</div>
`;

exports[`Welcome Welcome shows if failed 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation6 css-1p3yn26-MuiPaper-root"
>
  <div
    className="css-1y48f0p-MuiStack-root"
  >
    <span
      className="MuiTypography-root MuiTypography-button css-1xjj6hs-MuiTypography-root"
    >
      Welcome to poetry-parser!
    </span>
    <p
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Begin explorating your poetry.lock -file by uploading file.
    </p>
    <a
      className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
      href="/upload"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
    >
      Upload
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </a>
  </div>
</div>
`;
